### PR TITLE
feat(panel): Add fullscreen config option to the panel.

### PR DESCRIPTION
**Pull request by KarenParker**
@ErinCoughlan @gmoothart @DerekLouie @jelbourn 

Please review.

Fixes issue #7684.

### CODE REVIEW

File: src/components/panel/panel.js [hunk 1]
```diff
@@ -119,6 +119,9 @@ angular
  *   - `focusOnOpen` - `{boolean=}`: An option to override focus behavior on
  *     open. Only disable if focusing some other way, as focus management is
  *     required for panels to be accessible. Defaults to true.
+ *   - `fullScreen` - `{boolean=}`: Whether the panel should be full screen.
```
Comment (DevVersion) [inline src/components/panel/panel.js]: Why is that here camel-cased, and below just lowercase?

Comment (KarenParker) [inline src/components/panel/panel.js]: Done.

**Comment by jelbourn** [comment]
LGTM

**Comment by ErinCoughlan** [comment]
LGTM
